src/cmd/create.go

```diff
@@ -168,10 +176,43 @@ func create(cmd *cobra.Command, args []string) error {
 		containerArg = "--container"
 	}
 
+	image := createFlags.image
+
+	if cmd.Flag("file").Changed {
+		if !utils.PathExists(createFlags.file) {
```

Comment:
I see a potential solution. Copy the contents of the file to a temporary location or into a buffer and pass that to `podman build` instead.

Reply:
Yes.  I fiddled with `podman build` a bit more today.  This works:
```
$ cat /path/to/build/context/Containerfile | podman build --file - /path/to/build/context
```

We could try to spot the Containerfile in the build context, then [open](https://pkg.go.dev/os#Open) it to get a [io.Reader](https://pkg.go.dev/io#Reader) and feed it to `shell.Run(...)` as `stdin` when invoking `podman build ...`.

It's worth bearing in mind that `podman build` supports specifying the build context directory as a HTTP URL of an archive, Git repository or a Containerfile, Containerfile.in that's supposed to be pre-processed by `cpp(1)`, multiple Containerfiles and build contexts through a mix of `--file`, `--build-context` options and a `[context]` argument, etc..

Reading the code of the `BuildDockerfiles()` function in [github.com/containers/buildah/imagebuildah](https://github.com/containers/buildah/tree/main/imagebuildah), `TempDirForURL()` function in [](https://github.com/containers/buildah/blob/main/define), and the [podman build](https://github.com/containers/podman/blob/main/cmd/podman/images/build.go) code leading up to them, I think we should only support a simple subset of what `podman build` offers.  It will help us avoid fronting the entire `podman build` CLI in `toolbox create`, keep our error handling clean, and avoid race conditions that could lead to future (including security) bugs.

So, I propose supporting only a single build context that should contain a Container/Dockerfile.  Supporting HTTP URLs or the context looks simple enough, but we should avoid the rest.